crates/biome_cli/src/commands/mod.rs

```diff
@@ -220,6 +220,11 @@ pub enum BiomeCommand {
         #[bpaf(external(partial_graphql_linter), optional, hide_usage, hide)]
         graphql_linter: Option<PartialGraphqlLinter>,
 
+        // TODO: The flag requires a value here. Am I able to let them pass `--write-suppressions`,
+        // without a value? Option<Option<String>> wasn't it.
+        #[bpaf(long("write-suppressions"))]
+        write_suppressions: Option<String>,
```

Comment:
If you want to be able to pass both `--write-suppressions` to get `Some("default_val")` and `--write-suppressions=something to get `Some("something")` then yea, what you found is the right value.

Make a function `fn write_suppressions() -> impl Parser<Option<String>>` with code from the last comment (for `bar`), and do something like

```rust
        #[bpaf(external)]
        write_suppressions: Option<String>,
```

> I'd prefer if someone more familiar with bpaf weighed in.

:eyes:

Reply:
Ended up going with a different API. Thanks for the tips, though! 🧠

Reply:
```rust
    } else if !suppress && suppress_reason.is_some() {
        return Err(CliDiagnostic::IncompatibleEndConfiguration(
            IncompatibleEndConfiguration {
                reason: "--suppress-reason must be used with --suppress".to_string(),
            },
        ));
    }
```

This is one of the patterns `bpaf` is designed to prevent. You get your data from the parser and data are ready to be used without having to do additional validation and/or unwraps. As far as I understand it out of 4 possible combinations of those flags 3 are valid:

- `--suppress` alone
- `--suppress` + `--suppress-reason`
- neither is present

Natural representation for this in Rust is `Option<Option<String>>`, or inner option can be replaced with a custom enum - this reason + all reasons(?).

So

```rust
#[derive(Bpaf)]
enum Reason {
    WithReason {
        /// help here
        #[bpaf(long("suppress"))]
        _flag: (),
        suppress_reason: Option<String>,
    }
    /// help here (same as before)
    #[bpaf(long("suppress"))]
    NoReason
}
```

And use that as

```rust
    ...
    #[bpaf(external(reason))]
    write_suppressions: Option<Reason>
```

You can make enum a bit more natural with combinatoric API, but in practice this should be good enough.